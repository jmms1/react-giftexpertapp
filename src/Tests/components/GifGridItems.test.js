import React from 'react';
import { shallow } from 'enzyme';
import {GifGridItem} from '../../components/GifGridItem';


describe('Pruebas en GifItem', () => {

    const title = 'Un titulo'; 
    const url = 'http://localhost/algo.jpg';
    const wrapper = shallow(<GifGridItem
        title= {title}
        url = {url}
        />)
    
    test('should display component', () => {
        
        expect( wrapper ).toMatchSnapshot();
    })

    test('Debe contener un parrafo con el titulo', () => {

        const p = wrapper.find('p');
        expect (p.text().trim() ).toBe( title );
        
    })

    test('Debe contener una imagen con url y alt', () => {

        const img = wrapper.find('img');
        expect (img.prop('src')). toBe( url );
        expect (img.prop('alt')). toBe( title );
        
    })

    test(' debe tener clase css animate__bounce', () => {
        
        const div = wrapper.find('div');
        expect( div.hasClass('animate__bounce')).toBe(true);
    })
    
    
    
})
