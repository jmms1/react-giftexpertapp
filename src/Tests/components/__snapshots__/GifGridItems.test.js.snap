// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifItem should display component 1`] = `
<div
  className="card animate__animated animate__bounce"
>
  <img
    alt="Un titulo"
    src="http://localhost/algo.jpg"
  />
  <p>
    Un titulo
  </p>
</div>
`;
